### PR TITLE
Excelsior KS-23

### PR DESCRIPTION
**Pull request by Skegal**
## About The Pull Request
Adds in the KS-23 Shotgun for Excelsior, appears in the Means of Revolution disk with slug/buckshot/beanbag shells. also its own disk with the same rarity as the maxim.
stat wise is a mix between the regulator and the kammerer

## Why It's Good For The Game

Adds a new loadout option for excelsior agents for taking down and implanting target.
KS23Showcase

## Testing
Tested, runs perfectly fine.

## Changelog
:cl:
add: KS-23, KS-23 disk
imageadd: added KS-23.dmi
code: added ks23.dm, modified excelsior.dm/guns.dm
config: added the ks-23 to cev_eris.dme
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/shotgun/ks23.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/obj/item/gun/projectile/shotgun/pump/ks
+	name = "KS-23"
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/shotgun/ks23.dm:2]:
```suggestion
	name = "Excelsior SG KS-23"
```

File: code/modules/projectiles/guns/projectile/shotgun/ks23.dm [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	desc = "Karabin Spetsnaliy, desgined by Excelsior for a cheap crowd control for their agents, made with refused anti-aircraft barrels"
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/shotgun/ks23.dm:3]:
```suggestion
	desc = "Karabin Spetsialniy, designed by Excelsior for cheap crowd control for their agents, made with reuused anti-aircraft barrels."
```
Do we even consider aircraft existing?

File: code/modules/projectiles/guns/projectile/shotgun/ks23.dm [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	icon_state = "ks23"
+	item_state = "ks23"
+	max_shells = 4
+	w_class = ITEM_SIZE_HUGE
+	force = WEAPON_FORCE_ROBUST
+	slot_flags = SLOT_BACK
+	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_PLASTIC = 10, MATERIAL_STEEL = 15)
+	proj_step_multiplier = 1.2
+	ammo_type = /obj/item/ammo_casing/shotgun/pellet/scrap
+	price_tag = 3000
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/shotgun/ks23.dm:14]: Are you sure this is a right price for that matter cost?

File: code/game/objects/items/weapons/design_disks/excelsior.dm [hunk 4]
```diff
@@ -178,3 +182,18 @@
 		/datum/design/autolathe/ammo/maxim,							//Maxim ammo
 		/datum/design/autolathe/ammo/maxim/rubber,
 	)
+
+/obj/item/computer_hardware/hard_drive/portable/design/excelsior/ks
+	disk_name = "Excelsior - KS-23"
```
Comment (Firefox13) [inline code/game/objects/items/weapons/design_disks/excelsior.dm:187]:
```suggestion
	disk_name = "Excelsior - KS-23 SG"
```

File: code/datums/autolathe/guns.dm [hunk 5]
```diff
@@ -104,6 +104,10 @@
 	name = "SA SG \"Bojevic\""
 	build_path = /obj/item/gun/projectile/shotgun/bojevic
 
+/datum/design/autolathe/gun/ks23
+	name = "KS-23"
```
Comment (Firefox13) [inline code/datums/autolathe/guns.dm:108]:
```suggestion
	name = "Excelsior SG KS-23"
```

**Review by Firefox13 — COMMENTED** [review]
Make sure to follow the naming rules

File: code/modules/projectiles/guns/projectile/shotgun/ks23.dm (hunk 2 above)
Comment (Skegal) [reply code/modules/projectiles/guns/projectile/shotgun/ks23.dm:3]: not reused, and i wrote too fast i wanted rejected, i can change the entire description if you want

Comment (thedarkcrusader) [reply code/modules/projectiles/guns/projectile/shotgun/ks23.dm]: > Do we even consider aircraft existing? I suppose there are some on terrestial planets, but Excelsior aren't really fond of them

Looking at expeditionary fleet equipment snd vehicle page they do exist

File: code/modules/projectiles/guns/projectile/shotgun/ks23.dm (hunk 3 above)
Comment (Skegal) [reply code/modules/projectiles/guns/projectile/shotgun/ks23.dm:14]: how about 2000? the gladstone is 1800 for less material (i didnt put much though on the gun prices but thinking about it if the guild get their hands on it they could do a lot of profit indeed)

Comment (DiaFRAME444) [reply code/modules/projectiles/guns/projectile/shotgun/ks23.dm]: Price also decreases spawn price and the such so there's a reason to increase price if its a good weapon.

Comment (Skegal) [reply code/modules/projectiles/guns/projectile/shotgun/ks23.dm]: the moment i lower the price to 2K i get told to get it up xd, i leave it at 2K for now and i will wait for more intel to adjust the price, thanks a lot for the help.

File: code/game/objects/items/weapons/design_disks/excelsior.dm [hunk 6]
```diff
@@ -178,3 +182,18 @@
 		/datum/design/autolathe/ammo/maxim,							//Maxim ammo
 		/datum/design/autolathe/ammo/maxim/rubber,
 	)
+
+/obj/item/computer_hardware/hard_drive/portable/design/excelsior/ks
+	disk_name = "Excelsior - KS-23 SG"
+	desc = {"The back has a machine etching:\n \
+	\"Riot Control!!\""}
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/weapons/design_disks/excelsior.dm:189]:
```suggestion
	desc = "The back has a machine etching: \"Riot Control!.\""
```

This should maybe fix linters

**Comment by Humonitarian** [comment]
Devdel here, honestly not sure what this brings to the game in particular. A high-flesh damage weapon to excel? Their .40 SMG excels (pun not intended) at it already, and they have all the sidearms like the makarov being damage-heavy to top it further.

When it comes purely to high burst damage, mosins are already strong single-shot weapons.

There's really no good niché or purpose this shotgun serves for Excelsior, especially without any special ammunition.

Otherwise the gun is fine, and the sprite looks cool.

**Comment by ValoTheValo** [comment]
> Devdel here, honestly not sure what this brings to the game in particular. A high-flesh damage weapon to excel? Their .40 SMG excels (pun not intended) at it already, and they have all the sidearms like the makarov being damage-heavy to top it further.
> 
> When it comes purely to high burst damage, mosins are already strong single-shot weapons.
> 
> There's really no good niché or purpose this shotgun serves for Excelsior, especially without any special ammunition.
> 
> Otherwise the gun is fine, and the sprite looks cool.

variety, excelsior dont have a shotgun of thier own, buckshot is good for defending something like the science shuttle or maint corridors where excelbases are. slugs is just a shitty mosin.

**Comment by MLGTASTICa** [comment]
Could give the excelsior shotgun a special charge fire-mode similar to the one that lasers use(for added base defense potential)

**Comment by DiaFRAME444** [comment]
If it proves underwhelming we can say this design was obtained by Technomancers as a reason to give them a shotgun, maybe.

> Could give the excelsior shotgun a special charge fire-mode similar to the one that lasers use(for added base defense potential)

Isn't Charge mode for projectile weapons broken?

**Comment by ValoTheValo** [comment]
> If it proves underwhelming we can say this design was obtained by Technomancers as a reason to give them a shotgun, maybe.
> 
> > Could give the excelsior shotgun a special charge fire-mode similar to the one that lasers use(for added base defense potential)
> 
> Isn't Charge mode for projectile weapons broken?

tried to give it to the type 90 but it broke so uhh. yeah

**Comment by thevandie** [comment]
> Devdel here, honestly not sure what this brings to the game in particular. A high-flesh damage weapon to excel? Their .40 SMG excels (pun not intended) at it already, and they have all the sidearms like the makarov being damage-heavy to top it further.
> 
> When it comes purely to high burst damage, mosins are already strong single-shot weapons.
> 
> There's really no good niché or purpose this shotgun serves for Excelsior, especially without any special ammunition.
> 
> Otherwise the gun is fine, and the sprite looks cool.a
> 
> PS: given project leader approval, this wouldn't be bad for drip reasons for excel. It's still another weapon to be cool with, even if it doesn't add gameplay value outside of that.

It's A FUCKING SHOTGUN. That's unique enough.

**Comment by thevandie** [comment]
I absolutely love this thing

File: cev_eris.dme [hunk 7]
```diff
@@ -2441,6 +2441,7 @@
 #include "code\modules\projectiles\guns\projectile\dartgun.dm"
 #include "code\modules\projectiles\guns\projectile\flaregun.dm"
 #include "code\modules\projectiles\guns\projectile\shotgun.dm"
+#include "code\modules\projectiles\guns\projectile\shotgun\ks23.dm"
 #include "code\modules\projectiles\guns\projectile\sniper.dm"
```
Comment (Trilbyspaceclone) [inline cev_eris.dme:2445]:
```suggestion
#include "code\modules\projectiles\guns\projectile\sniper.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\ks23.dm"
```
Wait shouldnt this be a line lower as well?
Did you mannully put this in or did DM mess it up on you?

File: cev_eris.dme [hunk 8]
```diff
@@ -2441,6 +2441,7 @@
 #include "code\modules\projectiles\guns\projectile\dartgun.dm"
 #include "code\modules\projectiles\guns\projectile\flaregun.dm"
 #include "code\modules\projectiles\guns\projectile\shotgun.dm"
+#include "code\modules\projectiles\guns\projectile\shotgun\ks23.dm"
```
Comment (Trilbyspaceclone) [inline cev_eris.dme:2444]: This should be in line 2501 not here, not sure if you mannully placed it or if DM messed it up

Comment (Skegal) [reply cev_eris.dme:2444]: manually placed, will change that now

**Comment by Voldirs** [comment]
As a unique feature it can be coded as hybrid gun that can also be loaded with a powercell in addition to regular ammo which gives either projectile speed boost either small EMP pulse on hit. So it's just not another shotgun with its disk as rare as maxim's one.

File: cev_eris.dme [hunk 9]
```diff
@@ -2499,6 +2498,7 @@
 #include "code\modules\projectiles\guns\projectile\shotgun\regulator.dm"
 #include "code\modules\projectiles\guns\projectile\shotgun\sawnoff.dm"
 #include "code\modules\projectiles\guns\projectile\shotgun\type21.dm"
+#include "code\modules\projectiles\guns\projectile\shotgun\ks23.dm"
```
Comment (hyperioo) [inline cev_eris.dme:2501]:
```
#include "code\modules\projectiles\guns\projectile\shotgun\bojevic.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\bull.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\doublebarrel.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\gladstone.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\handmade.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\pump.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\regulator.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\sawnoff.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\type21.dm"
#include "code\modules\projectiles\guns\projectile\shotgun\ks23.dm"
```
It looks like the alphabetic ordering has been broken 👀 

If you "activate" it with Byond Dream Maker, it should be automatically added at the right place.

**Review by hyperioo — CHANGES_REQUESTED** [review]
^

File: cev_eris.dme (hunk 9 above)
Comment (Skegal) [reply cev_eris.dme:2501]: oh i didnt payed attention of an alphabetic order, i can change it right now, i got told to put it in line 2501 some days earlier

Comment (Skegal) [reply cev_eris.dme]: corrected

**Review by hyperioo — APPROVED** [review]
I'll merge it like that for drip reason, just to let excel have their shotgun. Feel free to think about a small quirk to make it unique. 

For instance maybe it could come with an integrated flashlight now that #8048 has been merged

**Comment by Skegal** [comment]
> I'll merge it like that for drip reason, just to let excel have their shotgun. Feel free to think about a small quirk to make it unique.
> 
> For instance maybe it could come with an integrated flashlight now that #8048 has been merged

oh nice, good idea, will think about it, i got other projects for eris but i will add it to my to-do list, thanks a lot, was fun to learn sprite and a bit of code

**Pull request merged**